custom_components/localtuya/common.py

```diff
@@ -89,17 +93,23 @@ def __get_status(self):
                     raise ConnectionError("Failed to update status .")
 
     def set_dps(self, state, dps_index):
-        # _LOGGER.info("running def set_dps from cover")
-        """Change the Tuya switch status and clear the cache."""
-        self._cached_status = ""
-        self._cached_status_time = 0
+        """Change value of a DP of the Tuya device and update the cached status."""
+        # _LOGGER.info("running def set_dps from TuyaDevice")
+        # No need to clear the cache here: let's just update the status of the
+        # changed dps as returned by the interface (see 5 lines below)
+        # self._cached_status = ""
+        # self._cached_status_time = 0
         for i in range(5):
             try:
-                return self._interface.set_dps(state, dps_index)
-            except Exception:
+                result = self._interface.set_dps(state, dps_index)
+                self._cached_status["dps"].update(result["dps"])
+                signal = f"localtuya_{self._interface.id}"
+                async_dispatcher_send(self._hass, signal, self._cached_status)
```

Comment:
Shouldn't it be `self._cached_status["dps"]` that is passed on? The expected format at the receiving end is just a dict with DPs.

Reply:
Nope, it actually works like this.
Also here:
`status = await hass.async_add_executor_job(device.status)`
you dispatch the entire status, not only the "dps" part.

Reply:
...and in fact, in the dps() method, you use:
`value = self._status["dps"].get(str(dps_index))`

Reply:
Ha, look at that. The I was thinking one thing but did something else! The we are ready to merge this!

Reply:
Yeah, and I think it's better like this because in the full status structure we have the "t" timestamp which might be useful for ignoring old messages maybe (I doubt we'll ever need this, but let's keep this possibility).

Reply:
I don't think we will ever need it. It's just used as a nounce due to AES ECB to mitigate replay attacks. Classic pattern.